### ISSUE TITLE
Error suppression operator breaks function behaviour

### ISSUE THREAD

**Issue by uncle-fed** [jqlang/jq#3018]
**Describe the bug**

Living for many years with jq 1.6 on RedHat Enterprise Linux 7, lots of jq code in use. 
After upgrading a fleet of hosts to RedHat Enterprise Linux 8 we started having jq producing bogus results.
The problem was narrowed down to how the error suppression operator `?` behaves after the OS upgrade.
The jq version remained at 1.6 after the OS upgrade but the actual compiled binary has changed (since it comes with the OS base repositories which are completely different to the previous OS version).

**To Reproduce**

` jq -n '["001","x2","003"] | map(tonumber? // .)'`

**Expected output**

This is how the jq 1.6 on RHEL7 behaved (and this is what is expected and behaves same way in all other environments and OSes, where we tried it).

`[1,"x2",3]`

**Actual result**

This is what jq 1.6 produces on RHEL8. In other words, it seems like as soon as the error suppression kicks in, the subsequent calls to the same function look as if the function is not performing properly (or at all).

`[1,"x2","003"]`

**Environment (please complete the following information):**

- jq version 1.6
- properly working on RHEL7
- giving bad result on RHEL8

**Additional context**

I've found this: https://github.com/jqlang/jq/issues/1885, https://github.com/jqlang/jq/issues/1859  (and related issues). However, it's nearly obvious that it can't be pure jq bug, since the same released jq version 1.6 worked fine for some years before RHEL8 was even conceived as an OS.  I just thought that before I go to RedHat with this, I might get some valuable information / advice on where to look for an answer. So, if there indeed was a similar / related bug fixed long time ago could it be that RedHat somehow did not incorporate the fix into their compiled code base? I have a suspicion that this kind of thing [happened before](https://bugzilla.redhat.com/show_bug.cgi?id=2008717). I am just not entirely certain how come new bugs appear in the jq version which is deemed to be the _same version_ as in previous major OS release, while clearly the behaviour is not the same. Does it mean that the bug was introduced in jq code since the original 1.6 release or does it mean that somehow Red Hat took older jq code for the newer OS release??

**Comment by emanuele6** [comment]
We don't maintain `jq 1.6`, nor the RHEL `jq` package, so you should probably report this problem elsewhere (e.g. to RHEL support), because we cannot do much!

**Comment by uncle-fed** [comment]
Thanks for the comment. I fully agree that this not a current bug in the current jq version. I was just hoping that somebody might point out that this kind of buggy behaviour is caused by some older bug which is known (and the fix for it is known too), because I could not directly find exactly the same bug reported. The closest I came in my search was [this issue](https://github.com/jqlang/jq/issues/1859) and my question is basically, is it the same issue or not? This would help me greatly if I could go to Red Hat and tell them that their currently packaged version is suffering from the jq issue "X" which was fixed in jq commit "Y".  Any insight would be appreciated!
